lb_content_resolver/fuzzy_index.py

```diff
@@ -0,0 +1,85 @@
+import os
+from time import time
+import datetime
+import re
+import sys
+
+from sklearn.feature_extraction.text import TfidfVectorizer
+import nmslib
+from unidecode import unidecode
+
+
+def ngrams(string, n=3):
+    """ Take a lookup string (noise removed, lower case, etc) and turn a list of trigrams """
+
+    string = ' ' + string + ' '  # pad names for ngrams...
+    ngrams = zip(*[string[i:] for i in range(n)])
+    return [''.join(ngram) for ngram in ngrams]
+
+
+class FuzzyIndex:
+    ''' 
+       Create a fuzzy index using a Term Frequency, Inverse Document Frequency (tf-idf)
+       algorithm. Currently the libraries that implement this cannot be serialized to disk,
+       so this is an in memory operation. Fortunately for our amounts of data, it should
+       be quick to rebuild this index.
+    '''
+
+    def __init__(self, index_dir):
+        self.index_dir = index_dir
+        self.vectorizer = None
+        self.index = None
+
+    def create(self):
+        try:
+            os.mkdir(self.index_dir)
+        except OSError as err:
+            print("Could not create index directory: %s (%s)" % (self.index_dir, err))
+            return
+
+    def encode_string(self, text):
+        return unidecode(re.sub(" +", "", re.sub(r'[^\w ]+', '', text)).strip().lower())
```

Comment:
Is this the same magic incantation as before? If so perhaps we should move it to a common location.

Reply:
It should only live in fuzzy index. removed.